### PR TITLE
build(aio): fix up API doc-gen templates

### PR DESCRIPTION
**Pull request by petebacondarwin**
* Remove whitespace before type specifiers
* Generate `new` and `call` member info for interfaces

### CODE REVIEW

File: aio/tools/transforms/templates/api/includes/class-overview.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {% macro renderMember(member) %}{% if not member.internal -%}
-<a class="code-anchor" href="#{$ member.name $}">{$ member.name $}</a> {$ params.paramList(member.parameters) | indent(4, false) | trim() $}{$ params.returnType(member.returnType) $}
+<a class="code-anchor" href="#{$ member.name $}">{$ member.name $}</a>{$ params.paramList(member.parameters) | indent(4, false) | trim() $}{$ params.returnType(member.returnType) $}
```
Comment (gkalpak) [inline aio/tools/transforms/templates/api/includes/class-overview.html]: OOC, is `... | trim` and `... | trim()` the same?

File: aio/tools/transforms/templates/api/includes/members.html [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
+    </div>
+    {% if doc.members.length or doc.callMember %}<hr>{% endif %}
+  {% endif %}
+  {% if doc.callMember %}
+    <div class="call-member">
+      <a id="{$ doc.callMember.name $}"></a>
+      <code-example hideCopy="true">{$ doc.callMember.name $}{$ params.paramList(doc.callMember.parameters) | trim $}{$ params.returnType(doc.callMember.returnType) $}</code-example>
+      {% if not doc.callMember.notYetDocumented %}{$ doc.callMember.description | marked $}{% endif %}
+    </div>
+    {% if doc.members.length %}<hr>{% endif %}
```
Comment (gkalpak) [inline aio/tools/transforms/templates/api/includes/members.html]: What is all members are internal?

File: aio/tools/transforms/templates/api/includes/members.html [hunk 3]
```diff
@@ -1,19 +1,29 @@
...
+{% if doc.members.length or doc.newMember or doc.callMember %}
 <section class="member-members">
   <h2>Members</h2>
+  {% if doc.newMember %}
+    <div class="new-member">
+      <a id="{$ doc.newMember.name $}"></a>
+      <code-example hideCopy="true">{$ doc.newMember.name $}{$ params.paramList(doc.newMember.parameters) | trim $}{$ params.returnType(doc.newMember.returnType) $}</code-example>
+      {% if not doc.newMember.notYetDocumented %}{$ doc.newMember.description | marked $}{% endif %}
+    </div>
+    {% if doc.members.length or doc.callMember %}<hr>{% endif %}
```
Comment (gkalpak) [inline aio/tools/transforms/templates/api/includes/members.html]: What is all members are internal?

File: aio/tools/transforms/templates/api/includes/members.html [hunk 4]
```diff
@@ -1,19 +1,29 @@
...
-      {$ member.description | marked $}
-      {% endif %}
+      {% if not member.notYetDocumented %}{$ member.description | marked $}{% endif %}
     </div>
-
-    {% if not loop.last %}
-    <hr>
-    {% endif %}
-
+    {% if not loop.last %}<hr>{% endif %}
```
Comment (gkalpak) [inline aio/tools/transforms/templates/api/includes/members.html]: What if all remaining members are internal?

**Review by gkalpak — APPROVED** [review]
A couple of nitpicks, but LGTM anyway.

Another nit: There is at least one more extra space between the class name and the opening `{` in [class overview][1], when `doc.heritage` is empty (e.g. [here][2]). Same for interface overview.

[1]: https://github.com/angular/angular/pull/17141/files#diff-fb5611d8a8b516e7dcf87a9795414d0cR8
[2]: https://pr17141-bb3f3354392eefb6aa7ae824fadc8568ce56bb56.ngbuilds.io/api/animations/AnimationBuilder#overview

**Comment by petebacondarwin** [comment]
If all the members are internal then it'll look rubbish :-)

**Comment by petebacondarwin** [comment]
I fixed the double space before heritage problem for both interfaces and classes.

**Pull request merged**